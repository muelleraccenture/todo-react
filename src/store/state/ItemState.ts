export interface ItemState {
    readonly label: string,
    readonly complete: boolean,
}
