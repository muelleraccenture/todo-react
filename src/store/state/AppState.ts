import {ItemState} from "./ItemState";

export interface AppState {
    readonly items: ReadonlyArray<ItemState>
}
