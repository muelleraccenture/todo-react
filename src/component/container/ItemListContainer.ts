import {connect, MapDispatchToPropsFunction, MapStateToProps} from "react-redux";
import {ItemList, ItemListDispatchProps, ItemListProps} from "../presentation/ItemList";
import {AppState} from "../../store/state/AppState";

export const mapStateToProps: MapStateToProps<ItemListProps, void, AppState> = (state) => ({
    items: state.items
})

export const mapDispatchToProps: MapDispatchToPropsFunction<ItemListDispatchProps, void> = (dispatch) => ({

})

export const ItemListContainer = connect(mapStateToProps, null)(ItemList)
