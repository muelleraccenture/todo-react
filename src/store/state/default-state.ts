import {AppState} from "./AppState";

export const DEFAULT_STATE: AppState = {
    items: [
        {label: "join experimental noise band", complete: false},
        {label: "become bilingual", complete: true},
    ]
}
