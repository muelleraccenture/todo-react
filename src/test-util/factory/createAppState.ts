import {AppState} from "../../store/state/AppState";

export const createAppState = (partial: Partial<AppState>): AppState => {
    const defaults: AppState = {
        items: []
    }

    return {...defaults, ...partial}
}
