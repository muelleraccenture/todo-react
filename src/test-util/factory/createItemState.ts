import {ItemState} from "../../store/state/ItemState";

export const createItemState = (partial: Partial<ItemState> = {}): ItemState => {
    const defaults: ItemState = {
        label: "",
        complete: false,
    }

    return {...defaults, ...partial}
}
