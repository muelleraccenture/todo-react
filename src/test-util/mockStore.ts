import {AppState} from "../store/state/AppState";

export const mockStore = (state: AppState) => ({
    getState: () => state,
    dispatch: jest.fn(),
    subscribe: jest.fn(),
    replaceReducer: jest.fn(),
})
