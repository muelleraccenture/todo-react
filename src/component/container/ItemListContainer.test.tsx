import React from 'react';
import {ItemListContainer} from "./ItemListContainer";
import renderer, {ReactTestRenderer} from "react-test-renderer";
import {AppState} from "../../store/state/AppState";
import {Store} from "redux";
import {createAppState} from "../../test-util/factory/createAppState";
import {createItemState} from "../../test-util/factory/createItemState";
import {mockStore} from "../../test-util/mockStore";
import {Provider} from "react-redux";

describe(ItemListContainer.name, () => {
    let tree: ReactTestRenderer, store: Store<AppState>

    beforeEach(() => {

        store = mockStore(createAppState({
            items: [
                createItemState({label: "do laundry", complete: true})
            ]
        }))

        tree = renderer.create(
            <Provider store={store}>
                <ItemListContainer></ItemListContainer>
            </Provider>,
        )
    })

    it("presentation has not changed", () => {
        expect(tree.toJSON()).toMatchSnapshot()
    })
})
