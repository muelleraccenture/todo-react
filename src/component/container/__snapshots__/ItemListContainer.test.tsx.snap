// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`presentation has not changed 1`] = `
<div>
  <ul>
    <li
      style={
        Object {
          "backgroundColor": "#B4FCF7",
          "color": "pink",
          "listStyleType": "none",
          "marginTop": "10px",
        }
      }
    >
      <p>
        do laundry
      </p>
      <p />
    </li>
  </ul>
</div>
`;
